packages/bridge-ui/src/relayer-api/service.ts

```diff
@@ -116,7 +110,9 @@ class RelayerAPIService implements RelayerAPI {
         );
 
         const event = events.find(
-          (e) => e.args.message.owner.toLowerCase() === address.toLowerCase(),
+          (e) =>
+            e.args.message.owner.toLowerCase() === address.toLowerCase() &&
+            e.args.message.depositValue.eq(tx.message.depositValue),
```

Comment:
For transactions included within the same block we were picking the first event out of many which resulted in wrong data being shown in the transaction list. This fixes it.

Reply:
Maybe we could add that as a comment?

Reply:
I just realized that even the depositValue could be the exact same. So we need to also compare `msgHash`.